.github/workflows/build.yaml

```
@@ -13,6 +13,11 @@ defaults:
   run:
     shell: bash
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

```suggestion
concurrency:
  group: ci-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}
  cancel-in-progress: true
```

You should use this one otherwise you will get cancellations across PRs